src/szk.rs

```diff
@@ -0,0 +1,565 @@
+            );
+        }
+
+        let leader_decision = szk_typ
+            .decide(
+                &verifier,
+                l_jr_part.clone(),
+                h_jr_part.clone(),
+                l_jr_seed.clone(),
+            )
+            .unwrap();
+        assert!(!leader_decision, "Leader validated after proof mutation");
+
+        // test mutated input share
+        let mut mutated_input = leader_input_share.clone();
+        mutated_input[0] *=
+            T::Field::from(<T::Field as FieldElementWithInteger>::Integer::try_from(23).unwrap());
+        let mutated_query_share = szk_typ
+            .query(&mutated_input, l_proof_share.clone(), &verify_key, &nonce)
+            .unwrap();
+        let mut verifier = mutated_query_share.verifier;
+
+        for (x, y) in verifier.iter_mut().zip(h_query_share.clone().verifier) {
+            *x += y;
+        }
+
+        let (mutated_jr_seed, mutated_jr_part) = match mutated_query_share.joint_rand_seed_and_part
+        {
+            Some([jr_seed, jr_part]) => (Some(jr_seed), Some(jr_part)),
+            None => (None, None),
+        };
+        if let Ok(leader_decision) = szk_typ.decide(
+            &verifier,
+            mutated_jr_part.clone(),
+            h_jr_part.clone(),
+            mutated_jr_seed,
+        ) {
+            assert!(!leader_decision, "Leader validated after input mutation");
+        };
+
+        // test mutated proof share
+        let (mut mutated_proof, blind_and_joint_rand_part) = match l_proof_share {
+            SzkProofShare::Leader {
+                uncompressed_proof_share,
+                blind_and_joint_rand_part,
+            } => (uncompressed_proof_share.clone(), blind_and_joint_rand_part),
+            _ => (vec![], None),
+        };
+        mutated_proof[0] *=
+            T::Field::from(<T::Field as FieldElementWithInteger>::Integer::try_from(23).unwrap());
+        let mutated_proof_share = SzkProofShare::Leader {
+            uncompressed_proof_share: mutated_proof,
+            blind_and_joint_rand_part,
+        };
+        let l_query_share = szk_typ
+            .query(
+                &leader_input_share,
+                mutated_proof_share,
+                &verify_key,
+                &nonce,
+            )
+            .unwrap();
+        let mut verifier = l_query_share.verifier;
+
+        for (x, y) in verifier.iter_mut().zip(h_query_share.clone().verifier) {
+            *x += y;
+        }
+
+        let (mutated_jr_seed, _mutated_jr_part) = match l_query_share.joint_rand_seed_and_part {
+            Some([jr_seed, jr_part]) => (Some(jr_seed), Some(jr_part)),
+            None => (None, None),
+        };
+        if let Ok(leader_decision) = szk_typ.decide(
+            &verifier,
+            l_jr_part.clone(),
+            h_jr_part.clone(),
+            mutated_jr_seed,
+        ) {
+            assert!(!leader_decision, "Leader validated after proof mutation");
+        };
+    }
+
+    #[test]
```

Comment:
Add a test for `Count` as well so that we try with and without joint randomness.

Reply:
There is one, right? line 558

Reply:
Yup, you're right! Sorry :/